source/common/config/grpc_mux_impl.cc

```diff
@@ -8,14 +8,18 @@
 namespace Envoy {
 namespace Config {
 
-GrpcMuxImpl::GrpcMuxImpl(const envoy::api::v2::Node& node, Grpc::AsyncClientPtr async_client,
+GrpcMuxImpl::GrpcMuxImpl(const envoy::api::v2::core::Node& node, Grpc::AsyncClientPtr async_client,
                          Event::Dispatcher& dispatcher,
                          const Protobuf::MethodDescriptor& service_method)
     : node_(node), async_client_(std::move(async_client)), service_method_(service_method) {
   retry_timer_ = dispatcher.createTimer([this]() -> void { establishNewStream(); });
 }
 
 GrpcMuxImpl::~GrpcMuxImpl() {
+  // Hack to force linking of the service: https://github.com/google/protobuf/issues/4221
+  envoy::service::discovery::v2::AdsDummy dummy;
+  envoy::service::ratelimit::v2::RateLimitRequest rls_dummy;
```

Comment:
We shouldn't need rls_dummy here, right? I don't think it makes sense to include rls.pb.h here. And how come this wasn't an issue previously in general?

Reply:
I've discovered this by running envoy-static with ADS. It is missing ADS descriptor without this change.

Reply:
OK, can you remove `rls_dummy` from here and the rls.pb.h include? It should not be needed in this file.

Reply:
Done.

Reply:
i am confused here. Why is 'AdsDummy' needed here now? It was able to link properly earlier and why the new package need this? I ran in to the same issue with latest source, that it is not able to find the 'MethodDescriptor' for ADS and 'bazel clean' seems to have resolved that. I may be wrong but Can you double check on this once please?

Reply:
it links correctly but if you try to call ADS methods, the reflection lookup fails since proto descriptor is missing. there's another PR that does a similar fix, so it is happening consistently. adding a dummy forces linking the proto descriptor.

Reply:
ok.got it thanks.